### DISCUSSION TITLE
Will it be possible get result status in future?

### DISCUSSION THREAD

**Discussion by vitsadecky** [zytzagoo/smtp-validate-email#63] [Q&A]
Would it not be possible to get the reason for the error from the outside? I'm considering applying this package, but I have a feeling that I won't know outside the class the reason for the error - (some basicenum list) fot example:
 -  invalid email
 - smtp communication failed
 - timeout. 
 - 
 - Thanks in advance for your response :-)

**Comment by zytzagoo** [comment]
I don't fully understand what you mean, but if you expect that an "invalid email" (whatever that means) should throw an exception, that's something you'll need to do yourself (should be relatively trivial to do).

An email being valid or not is IMO a boolean, not an exception. If you want an exception, you can throw one in your code, based on the value of the boolean.

There are various exceptions being thrown at various points in the process of connecting/validating/testing a single email, but they are caught, acted upon (based on how the validator is configured using options) and not re-thrown (in order to be able to produce a boolean in the end). 

If you could/would write how you'd like/imagine the "outside" code to look like (which methods of the package you'd like to call, which results they'd contain etc.), that would help too. Then we can see what it would take to make the package behave as needed/desired (if doable).

Worst case, maybe logging could be extended to log all exceptions as they're happening into another array (keyed by email maybe?). Then have some `getLoggedExceptions()` method which would return an array of all exceptions encountered... This is just a wild guess without even looking at the code, I have no idea if it would work easy in all cases...

**Reply by vitsadecky** [reply]
@zytzagoo Thank you for your answering :-).

I'll try differently. I need to distinguish whether it has actually been verified and is invalid. So I need to distinguish:
- The email has been verified and is invalid
- email could not be verified - timeout, invalid sender, service is unavaialbel

I think both samples :point_up:  returns in email result `false`, but it hasnt be truth

Is some possibility, how detect these states?

**Reply by vitsadecky** [reply]
i think is not good idea catch excptions for example `NoConnectionException` etc. because result is lie, i dont know if email is valid or not. Next unexpected response : you catch `UnexpectedResponseException` and dont say truth outside from these class.  

What if, for example, there is a bad implementation of this class or the response changes? You will always return the result false and set the value of response according to: $ this-> no_comm_is_valid. This may not be revealed for long time
